PR: https://github.com/odoo/odoo/pull/182639

From: 5620a16d55091b0ed74feee694a02252e3d13c17
From: David Monnom (moda)
Date: 2024-10-16 07:50:26

Breaking data model changes scores: del:1 + add:0, change matches:
-    _inherit = ['product.product']

Total Changes: 40

[REF] point_of_sale, pos_*: replace product.product by product.template

Historically, the point of sale module used the product.product model to
represent products cards in PoS. A workaround has been done to handle
variants, attributes and pricelist.

Now the fact that we miss the product.template model in the PoS module
is a blocker for the future of the PoS module. Causing a lot of
performance issues and limitations.

This commit replace the use of product.product as product card by the
product.template model. This will allow us to use the product.template
with pricelist, attributes, variants and other features.

The old product.product models has been proxied to the product.template
each variable that isn't on the product.product model will be taken from
the product.template model.

closes odoo/odoo#182639

Taskid: 4134659
Related: odoo/enterprise#71752
Signed-off-by: Joseph Caburnay (jcb) <jcb@odoo.com>

================================= pseudo patch: =================================

--- a/addons/pos_discount/models/__init__.py
+++ b/addons/pos_discount/models/__init__.py
@@ -3,4 +3,4 @@
 
 from . import pos_config
 from . import res_config_settings
-from . import product_product
+from . import product_template

--- a/addons/pos_discount/models/product_product.py
+++ b/addons/pos_discount/models/product_template.py
@@ -1,8 +1,8 @@
 from odoo import models
 
 
-class ProductProduct(models.Model):
-    _inherit = ['product.product']
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
 
     def _load_pos_data(self, data):
         res = super()._load_pos_data(data)
